### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by pedrojoya**
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `main` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `main` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `main`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](https://classroom.github.com/help/leaving-feedback-in-github)”.
</details>

Subscribed: @Aldione

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public static void main(String[] args) throws InterruptedException{
+
+        Thread thread = new Thread(new Seconds());
+        thread.start();
+        //Wait 12 seconds to interrupted the thread
+        thread.join(12000);
+        //Thread.sleep(12000); //Estuve probando con sleep al principio
+        //TimeUnit.SECONDS.sleep(12); //Lo hice asi para probarlo, pero veo que con thread.join se hace mas optimo.
+        //Interrupted the thread
+        thread.interrupt();
```
Comment (pedrojoya) [inline src/Main.java:14]:
```suggestion
        // Primero el hilo principal se duerme durante 12 segundos, mientras los que el
        // hilo secundario está mostrando la hora.
        Thread.sleep(12000);
        // Después interrumpe al hilo (le dice que vaya terminando).
        thread.interrupt();
        // Después espera a que el hilo termine de ejecutarse (la interrupción no es inmediata,
        // recuerda que sólo le has dicho que vaya terminando).
        thread.join();
```

File: src/Seconds.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            hours = cal.get( Calendar.HOUR_OF_DAY );
+            minutes = cal.get( Calendar.MINUTE );
+            seconds = cal.get( Calendar.SECOND );
+
+            System.out.printf("%d:%d:%d\n",hours, minutes, seconds);
+            try {
+                TimeUnit.SECONDS.sleep(1);
+            } catch (InterruptedException e) {
+                System.out.println("I was interrupted while I was sleeping");
+                finished = true; //Si añado un return, no me saca el print ultimo, y con el isInterrupted no termina de ejecutar lo.
```
Comment (pedrojoya) [inline src/Seconds.java:27]:
```suggestion
                //finished = true; //Si añado un return, no me saca el print ultimo, y con el isInterrupted no termina de ejecutar lo.
                // Cuando se pone return ; se termina la ejecución del método correspondiente,
                // en este caso el run, y evidentemente no ejecuta el código que haya a continuación,
                // ni el que haya fuera del bucle. Es una salida anticipada de método.
                // La ventaja es que te ahorras definir la variable finished.
```

File: src/Seconds.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        while (!Thread.currentThread().isInterrupted() && !finished) {
+
+            //Se podria usar el dateformat, pero lo he usado asi por hacerlo de otra forma y no sea parecida a los ejemplos y ver si asi funciona de la misma manera.
+            Calendar cal = Calendar.getInstance();
+            hours = cal.get( Calendar.HOUR_OF_DAY );
+            minutes = cal.get( Calendar.MINUTE );
+            seconds = cal.get( Calendar.SECOND );
+
+            System.out.printf("%d:%d:%d\n",hours, minutes, seconds);
```
Comment (pedrojoya) [inline src/Seconds.java:22]:
```suggestion
            // usa mejor un formateador de fechas, como SimpleDateFormat o TimeFormatter de la api de fechas de java 8
            System.out.printf("%d:%d:%d\n",hours, minutes, seconds);
```

**Comment by Aldione** [comment]
Pedro, no se que he hecho, pero se me ha borrado todo :/

**Pull request merged**